pulp_2to3_migration/app/plugin/rpm/migrator.py

```diff
@@ -39,26 +58,135 @@ class RpmMigrator(Pulp2to3PluginMigrator):
     pulp2_content_models = {
         'rpm': RPM,
         'erratum': Errata,
+        'modulemd': Modulemd,
+        'modulemd_defaults': ModulemdDefaults,
     }
     pulp2_collection = 'units_rpm'
     pulp3_plugin = 'pulp_rpm'
-    pulp3_repository = RpmRepository
-    content_models = {
-        'rpm': Pulp2Rpm,
-        'erratum': Pulp2Erratum,
+    pulp3_repository = pulp3_models.RpmRepository
+    content_models = OrderedDict([
+        ('rpm', Pulp2Rpm),
+        ('erratum', Pulp2Erratum),
+        ('modulemd', Pulp2Modulemd),
+        ('modulemd_defaults', Pulp2ModulemdDefaults),
+    ])
+    mutable_content_models = {
+         'modulemd': Pulp2Modulemd,
+         'modulemd_defaults': Pulp2ModulemdDefaults,
     }
     importer_migrators = {
         'yum_importer': RpmImporter,
     }
     distributor_migrators = {
         'yum_distributor': RpmDistributor,
     }
+    #premigrate_hook = {
+    #    'modulemd_defaults': ''
+    #}
 
     @classmethod
     async def migrate_content_to_pulp3(cls):
         """
         Migrate pre-migrated Pulp 2 RPM plugin content.
         """
         first_stage = ContentMigrationFirstStage(cls)
-        dm = DeclarativeContentMigration(first_stage=first_stage)
+        dm = RPMDeclarativeContentMigration(first_stage=first_stage)
         await dm.create()
+
+class RPMDeclarativeContentMigration(DeclarativeContentMigration):
+    """
+    A pipeline that migrates pre-migrated Pulp 2 rpm content into Pulp 3.
+    """
+
+    def pipeline_stages(self):
+        """
+        Build a list of stages.
+
+        This defines the "architecture" of the content migration to Pulp 3.
+
+        Returns:
+            list: List of :class:`~pulpcore.plugin.stages.Stage` instances
+
+        """
+        pipeline = [
+            self.first_stage,
+            QueryExistingArtifacts(),
+            ArtifactSaver(),
+            QueryExistingContents(),
+            ContentSaver(),
+            InterrelateContent(),
+            RelatePulp2to3Content(),
+        ]
+
+        return pipeline
+
+
+class InterrelateContent(Stage):
+    """
+    Stage for relating Content to other Content.
+    """
+
+    async def run(self):
+        """
+        Relate each item in the input queue to objects specified on the DeclarativeContent.
+        """
+        import pydevd
+        pydevd.settrace('localhost', port=12345, stdoutToServer=True, stderrToServer=True)
+        async for batch in self.batches():
+            module_packages_batch = []
+            modular_packages_batch = []
+            with transaction.atomic():
+                for dc in batch:
+                    if type(dc.content) == pulp3_models.Modulemd:
+                        thru, packages = self.relate_package_to_module(dc)
+                        module_packages_batch.extend(thru)
+                        modular_packages_batch.extend(packages)
+
+                pulp3_models.Package.objects.bulk_update(objs=set(modular_packages_batch),
```

Comment:
I'm not sure why it is not set during packages migration.

Reply:
it is a pulp2 issue. we agreed to fix this in pulp2